modules/core/src/main/scala/com/caesars/tracing/ZTracer.scala

```diff
@@ -0,0 +1,85 @@
+package com.caesars.tracing
+
+import cats.data.NonEmptyList
+import io.janstenpickle.trace4cats.inject.EntryPoint
+import io.janstenpickle.trace4cats.model.*
+import io.janstenpickle.trace4cats.{ErrorHandler, Span}
+import zio.*
+import zio.interop.catz.*
+
+//TODO: find away to remove the option
+//We have the following problem:
+//In application code, we should be blissfully unaware of what spans have been created before and where we are in the structure
+//Specifically, you should not need to know whether you are creating a root span or a child span
+//Also, applications may have more than one root span
+
+//What we would like, is a ZTracer that always has a span, and can perform `context`, `setStatus`, `putAll` and `span` safely
+final case class ZTracer private (
+    private val currentSpan: FiberRef[Option[ZSpan]],
+    private val entryPoint: EntryPoint[Task]
+) { self =>
+  def context: UIO[SpanContext] =
+    currentSpan.get.map(_.fold(SpanContext.invalid)(_.context))
+
+  def span[R, E, A](
+      name: String,
+      kind: SpanKind = SpanKind.Internal,
```

Comment:
If a user is doing a span manually for their application then I would say this would be an internal span. Consider the use case where we're trying to trace the amount of time a call is spent doing a database query. If we're relying on library instrumentation (for example zhttp/sttp client) then I would say that would fall into the server/client category